metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/execution/QueueConsumer.java

```diff
@@ -100,63 +98,97 @@ public QueueConsumer(Channel rabbitmqConsumerChannel, String rabbitmqQueueName,
   public void handleDelivery(String consumerTag, Envelope rabbitmqEnvelope,
       AMQP.BasicProperties properties, byte[] body) throws IOException {
     String objectId = new String(body, StandardCharsets.UTF_8);
-    LOGGER.info("WorkflowExecution id: {} received from queue.", objectId);
+    LOGGER.info("workflowExecutionId: {} - Received from queue.", objectId);
+    LOGGER.info("workflowExecutionId: {} - Claiming workflow execution", objectId);
+    Pair<WorkflowExecution, Boolean> workflowExecutionClaimedPair = workflowExecutionMonitor
+        .claimExecution(objectId);
+    WorkflowExecution workflowExecution = workflowExecutionClaimedPair.getLeft();
+    final Boolean workflowClaimed = workflowExecutionClaimedPair.getRight();
 
+    boolean sendAck = true;
+    if (workflowExecution == null) {
+      // This execution no longer exists and we need to ignore it.
```

Comment:
It could also mean that an exception occurred retrieving/claiming it (see `WorkflowExecutionMonitor.claimExecution()`). Is this to be treated differently?

Reply:
No. I would say like before we are assuming that it will be picked up from the failsafe if that occurs